src/pages/dashboard/seller/AllProducts/allProducts.js

```diff
@@ -0,0 +1,38 @@
+import React, { useEffect } from "react";
+import ProductsTable from "../../../../components/viewProducts/ProductsTable.js";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  getCategoriesThunk,
+  getProductsThunk,
+} from "../../../../redux/features/actions/products.js";
+import { getProduct } from "../../../../redux/features/slices/products.js";
+import Spinner from "../../../../components/viewProducts/spinner.js";
+
+const AllProducts = () => {
+  const dispatch = useDispatch();
+  const token = localStorage.getItem("token");
+  const { loading, error, products, categories } = useSelector(getProduct);
+  useEffect(() => {
+    if (token) {
+      dispatch(getCategoriesThunk()).then(() => {
+        dispatch(getProductsThunk(token));
+      });
+    }
```

Comment:
What happens if there is no token?

Reply:
There is going to be a table of wishes on the seller side and reviews on the main buyer side